source/lex.tex

```diff
@@ -1454,64 +1473,64 @@
-have the same value.
-\end{example}
-Either the integer part or the fraction part (not both) can be omitted.
-Either the radix point or the letter \tcode{e} or \tcode{E} and
-the exponent (not both) can be omitted from a decimal floating-point literal.
-The radix point (but not the exponent) can be omitted
-from a hexadecimal floating-point literal.
-The integer part, the optional radix point, and the optional fraction part,
-form the \defn{significand} of the floating-point literal.
-In a decimal floating-point literal, the exponent, if present,
-indicates the power of 10 by which the significand is to be scaled.
-In a hexadecimal floating-point literal, the exponent
-indicates the power of 2 by which the significand is to be scaled.
-\begin{example}
-The floating-point literals \tcode{49.625} and \tcode{0xC.68p+2} have the same value.
-\end{example}
-If the scaled value is in
-the range of representable values for its type, the result is the scaled
-value if representable, else the larger or smaller representable value
-nearest the scaled value, chosen in an \impldef{choice of larger or smaller value of
-floating-point literal} manner.
-\indextext{literal!\idxcode{double}}%
-The type of a floating-point literal is \tcode{double}
 \indextext{literal!type of floating-point}%
-unless explicitly specified by a suffix.
 \indextext{literal!\idxcode{float}}%
 \indextext{suffix!\idxcode{F}}%
 \indextext{suffix!\idxcode{f}}%
-The suffixes \tcode{f} and \tcode{F} specify \tcode{float},
 \indextext{suffix!\idxcode{L}}%
 \indextext{suffix!\idxcode{l}}%
 \indextext{literal!\idxcode{long double}}%
-the suffixes \tcode{l} and \tcode{L} specify \tcode{long}
-\tcode{double}. If the scaled value is not in the range of representable
+The type of a \grammarterm{floating-point-literal} is determined by
+its \grammarterm{floating-point-suffix} as specified in \tref{lex.fcon.type}.
+\begin{simpletypetable}
+{Types of \grammarterm{floating-point-literal}{s}}
+{lex.fcon.type}
+{ll}
+\topline
+\lhdr{\grammarterm{floating-point-suffix}} & \rhdr{type} \\ \capsep
+none & \keyword{double} \\
+\tcode{f} or \tcode{F} & \keyword {float} \\
+\tcode{l} or \tcode{L} & \keyword{long} \keyword{double} \\
+\end{simpletypetable}
+
+\pnum
+\indextext{literal!floating-point}%
+The \defn{significand} of a \grammarterm{floating-point-literal}
+is the \grammarterm{fractional-constant} or \grammarterm{digit-sequence}
+of a \grammarterm{decimal-floating-point-literal}
+or the \grammarterm{hexadecimal-fractional-constant}
+or \grammarterm{hexadecimal-digit-sequence}
+of a \grammarterm{hexadecimal-floating-point-literal}.
+A \grammarterm{digit-sequence} is interpreted as a decimal (base ten) number.
+A \grammarterm{hexadecimal-digit-sequence} is interpreted
+as a hexadecimal (base sixteen) number.
+In a \grammarterm{decimal-floating-point-literal},
+the \grammarterm{exponent-part}, if present,
+indicates the power of 10 by which the significand is to be scaled.
```

Comment:
We don't say how to interpret the *exponent-part*. Maybe

> If an *exponent-part* or *binary-exponent-part* is present, the *exponent* of the *floating-point-literal* is the result of interpreting the sequence of an optional *sign* and the *digit*s as a base 10 integer. Otherwise, the exponent is 0. The *scaled value* is *s* * 10^*e* for a *decimal-floating-point-literal* and *s* * 2^*e* for a *hexadecimal-floating-point-literal*, where *s* is the significand and *e* is the exponent.

Reply:
Hyphenated base-10, actually introduced $s$ and used math-mode \times. Also replaced "the exponent is 0" with "the exponent $e$ is 0".

Reply:
Hyphenated base-10, actually introduced $s$ and used math-mode \times. Also replaced "the exponent is 0" with "the exponent $e$ is 0".